server/src/main/java/io/crate/expression/reference/sys/check/node/MaxShardsPerNodeSysCheck.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.reference.sys.check.node;
+
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.indices.ShardLimitValidator;
+
+@Singleton
+public class MaxShardsPerNodeSysCheck extends AbstractSysNodeCheck {
+
+    static final int ID = 8;
+    private static final String DESCRIPTION = "Maximum shards per node limit exceeded.";
```

Comment:
I believe currently this check should never trigger as CrateDB prevents new tables and partitions which would put the number of shards above the limit.

Reply:
Yes, correct this was only the first draft.